VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+  scheme-specified version comparison and ordering.
+
+
+Some of the known versioning schemes
+----------------------------------------
+
+These are a few known versioning schemes for some common Package URL
+`types` (aka. ``ecosystem``).
+
+- **deb**: Debian and Ubuntu https://www.debian.org/doc/debian-policy/ch-relationships.html
+  The comparators are <<, <=, =, >= and >>.
+
+- **rpm**: RPM distros https://rpm-software-management.github.io/rpm/manual/dependencies.html
+  The a simplified rmpvercmp version comparison routine is used by archlinux Pacman.
+
+- **gem**: Rubygems https://guides.rubygems.org/patterns/#semantic-versioning
+  which is similar to ``node-semver`` for its syntax, but does not use semver
+  versions.
+
+- **npm**: npm uses node-semver which is based on semver with its own range
+  notation https://github.com/npm/node-semver#ranges
+  A similar but different scheme is used by Rust
+  https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html
+  and several other package types may use ``node-semver``-like ranges. But most
+  of these related schemes are not strictly the same as what is implemented in
+  ``node-semver``. For instance PHP ``composer`` may need its own scheme as this
+  is not strictly ``node-semver``.
+
+- **pypi**: Python https://www.python.org/dev/peps/pep-0440/
+
+- **cpan**: Perl https://perlmaven.com/how-to-compare-version-numbers-in-perl-and-for-cpan-modules
+
+- **golang**: Go modules https://golang.org/ref/mod#versions use ``semver`` versions
+  with a specific minimum version resolution algorithm.
+
+- **maven**: Apache Maven supports a math interval notation which is rarely seen
+  in practice http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html
+
+- **nuget**: NuGet https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges
+  Note that Apache Maven and NuGet are following a similar approach with a
+  math-derived intervals syntax as in https://en.wikipedia.org/wiki/Interval_(mathematics)
+
+- **gentoo**: Gentoo https://wiki.gentoo.org/wiki/Version_specifier
+
+- **alpine**: Alpine linux https://gitlab.alpinelinux.org/alpine/apk-tools/-/blob/master/src/version.c
+  which is using Gentoo-like conventions.
+
+- **generic**: a generic version comparison algorithm (which will be specified
+  later, likely based on a split on any wholly alpha or wholly numeric segments
+  and dealing with digit and string comparisons, like is done in libversion)
+
+
+TODO: add Rust, composer and archlinux, nginx, tomcat, apache.
+
+A separate document will provide details for each versioning scheme and:
+
+- how to convert its native range notation to the ``vers`` notation and back.
+- how to compare and sort two versions in a range.
+
+This versioning schemes document will also explain how to convert CVE and OSV
+ranges to ``vers``.
+
+
+Implementations
+-----------------------
+
+- Python: https://github.com/nexB/univers
+- Yours!
```

Comment:
May be of interest to some, I started building a Java implementation at https://github.com/nscuro/versatile

Capable of composing, parsing, simplifying, and validating. Supports a few major versioning schemes already, but coverage is not as good as that of univers yet. :)